packages/argo-checkout/src/extension-points/api/post-purchase/post-purchase.ts

```diff
@@ -109,3 +102,111 @@ interface Metafield {
   value: string | number;
   valueType: 'integer' | 'string' | 'json_string';
 }
+
+interface ExplicitDiscount {
+  value: number;
+  valueType: ExplicitDiscountType;
+  title: string;
+}
+
+interface AddVariantChange {
+  type: ChangeType.AddVariant;
+  variantId: number;
+  quantity: number;
+  discount?: ExplicitDiscount;
+}
+
+type Changes = AddVariantChange[];
+
+interface Changeset {
+  changes: Changes;
+}
+
+interface Money {
+  amount: string;
+  /** In ISO 4217 format */
+  currencyCode: string;
+}
+
+/** Represents an amount in both shop and presentment currencies. */
+interface MoneyBag {
+  shopMoney: Money;
+  presentmentMoney: Money;
+}
+
+interface LineItem {
+  /** The discounted price per quantity. */
+  discountedPriceSet: MoneyBag;
+  /** The price per quantity */
+  priceSet: MoneyBag;
+}
+
+interface AddedTaxLine {
+  /** The tax amount */
+  priceSet: MoneyBag;
+  rate: string;
+  title: string;
+}
+
+interface UpdatedLineItem {
+  /** The discounted price per quantity. */
+  discountedPriceSet: MoneyBag;
+  /** The price per quantity */
+  priceSet: MoneyBag;
+  productId: number;
+  variantId: number;
+  productHandle: string;
+  /** How many items are being purchased in this line.*/
+  quantity: number;
+}
+
+interface UpdatedPurchase {
+  /** Updated total price of the purchase with discounts but before shipping, taxes, and tips. */
+  subtotalPriceSet: MoneyBag;
+  /** Updated final price of the purchase */
+  totalPriceSet: MoneyBag;
+  addedTaxLines: AddedTaxLine[];
+  updatedLineItems: UpdatedLineItem[];
+  /** The amount left unpaid after the update */
+  totalOutstandingSet: MoneyBag;
+}
+
+interface ChangesetError {
+  code: string;
+  message: string;
+}
+
+type ChangesetResult =
+  | {
+      errors: ChangesetError[];
+      status: ChangesetProcessingStatus.Unprocessed;
+      updatedPurchase: UpdatedPurchase | null;
+    }
+  | {
+      errors: ChangesetError[];
+      status:
+        | ChangesetProcessingStatus.Processed
+        | ChangesetProcessingStatus.PartiallyProcessed;
+      updatedPurchase: UpdatedPurchase;
+    };
+
+export enum ChangesetProcessingStatus {
+  /** Changeset was successfully processed */
+  Processed = 'processed',
+  /**
+   * Changeset was processed, but not fully.
+   * Usually means the initial purchase was edited successfully, but payment failed - card didn't have enough funds, for example.
+   */
+  PartiallyProcessed = 'partially_processed',
+  /** Changeset was not processed because an error occurred - for example, when adding a variant that's out-of-stock. */
+  Unprocessed = 'unprocessed',
+}
+
+export enum ExplicitDiscountType {
+  Percentage = 'percentage',
+  FixedAmount = 'fixed_amount',
+}
+
+export enum ChangeType {
+  AddVariant = 'add_variant',
+}
```

Comment:
IMO, it's preferable to declare these as types (not enums) so they can traverse cases where `import type {...}` is used